mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp

```diff
@@ -673,25 +678,22 @@ extern "C" void mcpuMemset5DHalfRandInt(
     int64_t stride0, int64_t stride1, int64_t stride2, int64_t stride3,
     int64_t stride4, short min, short max, uint32_t seed) {
 
-  // Generate tables for converting float to fp16
-  unsigned short basetable[512];
-  unsigned char shifttable[512];
-  generateTables(basetable, shifttable);
-
   if (seed == 0)
     std::srand(time(0));
   else
     std::srand(seed);
 
-  float value;
+  float minf = min;
+  float maxf = max;
+
   for (unsigned i = 0; i < size0; ++i)
     for (unsigned j = 0; j < size1; ++j)
       for (unsigned k = 0; k < size2; ++k)
         for (unsigned l = 0; l < size3; ++l)
           for (unsigned m = 0; m < size4; ++m) {
-            value = randomIntegerValue(min, max);
+            float value = randomFloatValue(minf, maxf);
```

Comment:
This is INCORRECT. The signature of the function demands an integer.

Reply:
Could this explain the need for really high tolerances as well?

(and if for some reason a float value is needed, cast after rand()?)

Reply:
I wasn't aware of the context of high tolerances.

If a float value is needed, `@mcpuMemset5DHalfRandFloat` should be used instead of `@mcpuMemset5DHalfRandInt`

Reply:
randomFloatValue is the one with the 0.1 factor when minf and maxf are equal.  I bet that's making the difference, more than integer-vs-float.